File: BubbleSort.js
//BubbleSort.js

Array.prototype.BubbleSort = function(){
	var i, j, flag;
	console.log(this);
	var size = this.length;
	for(i = 0; i < size-1 ; i++){
		flag = 0;
		for(j = 0; j < size-1-i; j++){
			if(this[j] > this[j+1]){
				var tmp = this[j];
				this[j] = this[j+1];
				this[j+1] = tmp;
				flag = 1;
			}
		}
		if(!flag){
			break;
		}
	}
	return this;
}


var array = [1,2,3,4,5,6];
var array = [6,5,4,3,2,1];
console.log(array.BubbleSort());

File: BubbleSort.java
public class BubbleSort{
	public static void main(String[] args){
		// int [] array = { 2,5,6,23,1,10};
		int [] array = { 1,2,3,4,5,6};
		int i, j, flag, tmp;
		for(i = 0 ; i < array.length-1 ; i++){
			flag = 0;
			for(j = 0; j < array.length-1-i ; j++){
				System.out.println("비교");
				if(array[j] > array[j+1]){
					tmp = array[j];
					array[j] = array[j+1];
					array[j+1] = tmp;
					flag = 1;
				}
			}
			if(flag == 0)
				break;
		}

		for(int h = 0 ; h < array.length; h++){
			System.out.println(array[h]+ "");
		}

	}
}